catalyst/dl/experiment/config.py

```diff
@@ -323,32 +324,63 @@ def get_scheduler(self, stage: str, optimizer: Optimizer) -> Scheduler:
 
     @staticmethod
     def _get_transform(**params) -> Callable:
-        if "transforms" in params:
-            transforms_composition = [
-                ConfigExperiment._get_transform(**transform_params)
-                for transform_params in params["transforms"]
-            ]
-            params.update(transforms=transforms_composition)
+        key_value_flag = params.pop("_key_value", False)
 
-        transform = TRANSFORMS.get_from_params(**params)
+        if key_value_flag:
+            transforms_composition = {
+                key: ConfigExperiment._get_transform(**params_)
+                for key, params_ in params.items()
+            }
+
+            transform = {
+                key: Augmentor(
+                    dict_key=key,
+                    augment_fn=transform,
+                    input_key=key,
+                    output_key=key,
+                )
+                for key, transform in transforms_composition.items()
+            }
+        else:
+            if "transforms" in params:
+                transforms_composition = [
+                    ConfigExperiment._get_transform(**transform_params)
+                    for transform_params in params["transforms"]
+                ]
+                params.update(transforms=transforms_composition)
+
+            transform = TRANSFORMS.get_from_params(**params)
 
         return transform
 
     def get_transforms(self, stage: str = None, mode: str = None) -> Callable:
+        """
+        Returns transform for a given stage & mode
+
+        Args:
+            stage (str): stage name
+            mode (str): mode name, will be used only if the value of
+                `_key_value`` is ``True``
+        """
         transform_params = deepcopy(
-            self.stages_config[stage]["transform_params"].get(mode, {})
+            self.stages_config[stage].get("transform_params", {})
         )
 
+        key_value_flag = transform_params.pop("_key_value", False)
+        if key_value_flag:
+            transform_params = transform_params.get(mode, {})
+
         transform = self._get_transform(**transform_params)
-        if transform is not None:
-            def process(dict_):
-                result = transform(**dict_)
-                return result
-        else:
-            def process(dict_):
+        if transform is None:
+            def transform(dict_):
                 return dict_
+        elif not isinstance(transform, AugmentorCompose):
```

Comment:
I think this line is a bug. Using the transform.yml provided in _test_cv_segmentation, the resulting transform before this line is a Compose object and will be judged as NOT AugmentorCompose. Finally the returning transform cannot accept the dict with 'image' and 'mask' keys as input, thus leading to error. Without this judgment (e.g. removing the not), the transform will behave as anticipated.

Reply:
could you please make an issue,
here is the same code - https://github.com/catalyst-team/catalyst/blob/master/catalyst/dl/experiment/config.py#L415